Change sign-in flow pages as per new design

SudharakaP opened the pull request:
Resolve https://github.com/opencollective/opencollective/issues/5572

⚠️ Requires: https://github.com/opencollective/opencollective-api/pull/7592

https://user-images.githubusercontent.com/12435965/170743049-cf9c8060-275a-412a-aea0-1658e9cf65ba.mp4

Betree reviewed (commented):
Things I haven't tested yet (@SudharakaP you can already check if that works):
- 2FA
- Pages behaviors when already signed in

**Feedback**
- [x] I see no difference between `/create-account` and `create-accountv2`. Do we really need a separate route/component?
- [ ] The route https://opencollective-frontend-6uzyythwq-opencollective.vercel.app/create-accountv2 lands me on sign in, not sign up
- [ ] Footer should be hidden
- [x] Would be great to move error messages closer to the form, otherwise, they'll be difficult to see on mobile. I would actually advise to rather move to `addToast` for consistency with what we do elsewhere.
- [x] After you get an error, changing the email or moving from sign-in to create-account should reset the error
- [x] As it sounds we may want to reuse it in different places, I would rename:
`public/static/images/sign-in-page-logo.png` -> `public/static/images/oc-logo-watercolor-256.png` 
- [x] Implement "back" link on create-account page (see updated design)
- [x] Implement the single-query email lookup/sign-in
- [x] Email exists design mismatch
![image](https://user-images.githubusercontent.com/1556356/170989485-dd3d8bc9-bfe5-4882-a324-3575a7b4a40b.png)

Betree commented:
@SudharakaP I've updated by feedback above with the conclusions of today's discussion

SudharakaP commented:
> @SudharakaP I've updated by feedback above with the conclusions of today's discussion

Thanks and I've pushed changes to handle the issues you mentioned. 

>     * [ ]  The route https://opencollective-frontend-6uzyythwq-opencollective.vercel.app/create-accountv2 lands me on sign in, not sign up

For this one I have a feeling this is a Vercel problem such that the rewrites were not updated correctly. I tried this on my local and everything seems to work fine. Would you be able to clear the cache and redeploy on Vercel by any chance. Sorry I don't have access to the Vercel deployments.

SudharakaP commented on components/SignInV2.js:
```diff
@@ -0,0 +1,220 @@
+                    height={32}
+                    lineHeight={0}
+                    fontWeight="500"
+                    disabled={!email || error}
```
Here the disabling is done only when the email field is empty (and the user touched it) or there's an error in the email that is entered (which is shown below the email field). If we don't disable we will see the button being enabled for these two obvious cases (see screenshots below). To me it doesn't make sense to enable the button when the email field is empty; nor when the email field has something like one character which is obviously not an email address and we show the validation error below the field. For all other errors we show the toast after the user clicks on the `Continue` button. But if you disagree, I am happy to change this behavior to let the user click on the `Continue` regardless even in these two cases. 

![image](https://user-images.githubusercontent.com/12435965/171550851-8d0e76ee-24a4-4b57-b101-6c6d8fabba34.png)

![image](https://user-images.githubusercontent.com/12435965/171550885-b72f0f0b-b7f1-4c96-b934-2082be322b8c.png)

Betree replied:
Ref: https://uxplanet.org/disabled-buttons-in-user-interface-4dafda3e6fe7

> It is still possible to disable a submit button by default for login, shipping, billing, and many other forms when all fields are empty. However, when the user provides the information, it’s better to enable the Submit button. Let users make mistakes and provide appropriate feedback so that they can easily recover from error states.

So disabled if empty, yes. But I would enable it even if the email is not valid.

```suggestion
                    disabled={!email}
```

Betree commented:
> For this one I have a feeling this is a Vercel problem such that the rewrites were not updated correctly. I tried this on my local and everything seems to work fine. Would you be able to clear the cache and redeploy on Vercel by any chance. Sorry I don't have access to the Vercel deployments.

I can't see the URL for create-accountV2 anymore in `rewrites.js` did we get rid of it?

SudharakaP commented:
> > For this one I have a feeling this is a Vercel problem such that the rewrites were not updated correctly. I tried this on my local and everything seems to work fine. Would you be able to clear the cache and redeploy on Vercel by any chance. Sorry I don't have access to the Vercel deployments.
> 
> I can't see the URL for create-accountV2 anymore in `rewrites.js` did we get rid of it?

@Betree : This part is on the other PR; https://github.com/opencollective/opencollective-frontend/pull/7834

Betree reviewed (commented):
- Proptypes warning

![image](https://user-images.githubusercontent.com/1556356/172803310-bb8e258a-b719-4fb8-abc5-6fcfbb4926d4.png)


- When visiting `/signinv2` while authenticated, I see the form without any loading at the top right. I think we should show a loading state when `loadingLoggedInUser` is true.

![image](https://user-images.githubusercontent.com/1556356/172802670-dbfde12f-9961-483e-8b3c-e27ad3f1fb7c.png)

Related: when opening the sign-in link  with a token the form is flashing:

![image](https://user-images.githubusercontent.com/1556356/172804374-59823297-4014-4a29-8b37-fa97e955b6b0.png)

SudharakaP commented:
@Betree : Thanks for pointing these out. 

> * Proptypes warning

Fixed. 👍🏽 

>     * When visiting `/signinv2` while authenticated, I see the form without any loading at the top right. I think we should show a loading state when `loadingLoggedInUser` is true.

I have fixed it but on a slightly different note. Since on the in sign-in flow we don't show the profile or the change-log button, I have have made sure that as long as the page is still loading and the user is seeing the new sign-in page the profile or the change-log button is not shown. This is consistent with design; so as long as the user sees the new sign in page we show the header with nothing in it except the logo. Here's a screenshot. 

https://user-images.githubusercontent.com/12435965/172990728-82606920-11ba-4636-9375-38c793b30a10.mp4

Betree commented:
> @Betree : Thanks for pointing these out.
> 
> > * Proptypes warning
> 
> Fixed. 👍🏽
> 
> > ```
> > * When visiting `/signinv2` while authenticated, I see the form without any loading at the top right. I think we should show a loading state when `loadingLoggedInUser` is true.
> > ```
> 
> I have fixed it but on a slightly different note. Since on the in sign-in flow we don't show the profile or the change-log button, I have have made sure that as long as the page is still loading and the user is seeing the new sign-in page the profile or the change-log button is not shown. This is consistent with design; so as long as the user sees the new sign in page we show the header with nothing in it except the logo. Here's a screenshot.
> loading-signinv2.mp4

I don't understand how this addresses my feedback:
> When visiting /signinv2 while authenticated, I see the form without any loading at the top right. I think we should show a loading state when loadingLoggedInUser is true.

My point was if `loadingLoggedInUser` is true, it means we're probably going to redirect. There's no point in having a form flashing and we should display what's really happening: we're loading. Not showing the changelog button is not going to help, what we want is a loading spinner. a loading frame, or whatever helps for the user to understand that a request is going on and we're processing it. We therefore don't want them to fill a form, so it should not be displayed.

SudharakaP commented:
> My point was if `loadingLoggedInUser` is true, it means we're probably going to redirect. There's no point in having a form flashing and we should display what's really happening: we're loading. Not showing the changelog button is not going to help, what we want is a loading spinner. a loading frame, or whatever helps for the user to understand that a request is going on and we're processing it. We therefore don't want them to fill a form, so it should not be displayed.

Ops, sorry I misunderstood here. I've fixed this by making sure we are showing a `LoadingGrid` when the user is being re-directed; 

https://user-images.githubusercontent.com/12435965/173083782-22eb7693-da04-4756-a579-fb0fc20441ae.mp4

Let me know how it looks. We have the same thing happening in the older signin form, but I didn't add it there as we are going to deprecate it anyways once this new signin flow is underway.

znarf reviewed (changes requested):
This is not ready to merge, there are conflicts, E2E tests are not passing.

SudharakaP commented:
> This is not ready to merge, there are conflicts, E2E tests are not passing.

Sorry, missed that there were lang file conflicts. I've re-based and resolved them.

Betree commented:
Adding a commit as the legacy redirect for Vercel didn't include the status code (`pages/api/users/signin.js`), making it fail on deploy previews. We couldn't detect this locally with the current setup.

Betree commented on components/SignInV2.js:
```diff
@@ -0,0 +1,240 @@
+import StyledLink from './StyledLink';
+import { Span } from './Text';
+
+const SignInFooterLink = styled(StyledLink)`
```
These are internal links, we need to use the `Link` component to prevent a full page reload. I'm pushing a fix for this

znarf reviewed (approved):
We applied some small fixes. Good to go.

Betree reviewed (approved):
For followup issues:
- Pressing "Enter" with an invalid email should display the error

SudharakaP commented:
> We applied some small fixes. Good to go.

Thanks for the feedback and review. 

> For followup issues:
> 
>     * Pressing "Enter" with an invalid email should display the error

@Betree : Yes, I've corrected this in the other PR; https://github.com/opencollective/opencollective-frontend/pull/7834
